### PR TITLE
Switch from using bind.TransactOpts for transactions

### PR DESCRIPTION
**Pull request by vezenovm**
…ently still being used

## Related Issue Or Context

Closes: #49 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [ ] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: chains/evm/evmclient/evm-client.go [hunk 1]
```diff
@@ -125,6 +124,18 @@ func (c *EVMClient) LatestBlock() (*big.Int, error) {
 	return head.Number, err
 }
 
+func (c *EVMClient) IsEIP1559Activated() (bool, error) {
+	if head, err := c.HeaderByNumber(context.TODO(), nil); err != nil {
+		if head.BaseFee != nil {
+			return true, nil
+		} else {
+			return false, nil
+		}
```
Comment (waymobetta) [inline chains/evm/evmclient/evm-client.go:133]: Is this meant to be within the err block?

File: chains/evm/voter/proposal.go [hunk 2]
```diff
@@ -93,12 +94,24 @@ func (p *Proposal) Execute(client ChainClient) error {
...
-	tx := evmtransaction.NewTransaction(opts, p.BridgeAddress, big.NewInt(0), cId, input)
+
+	var tx evmclient.CommonTransaction
+	if activated, err := client.IsEIP1559Activated(); err != nil {
+		if activated {
+			tx = evmtransaction.NewDynamicFeeTransaction(cId, opts.Nonce.Uint64(), &p.BridgeAddress, big.NewInt(0), opts.GasTipCap, opts.GasFeeCap, opts.GasLimit, input)
+		} else {
+			tx = evmtransaction.NewTransaction(opts.Nonce.Uint64(), &p.BridgeAddress, big.NewInt(0), opts.GasLimit, opts.GasPrice, input)
+		}
+	} else {
```
Comment (waymobetta) [inline chains/evm/voter/proposal.go:112]: Is this also meant to be within the error block?

File: chains/evm/voter/proposal.go [hunk 3]
```diff
@@ -132,12 +143,24 @@ func (p *Proposal) Vote(client ChainClient) error {
...
 	opts.GasLimit = uint64(2000000)
-	tx := evmtransaction.NewTransaction(opts, p.BridgeAddress, big.NewInt(0), cId, input)
+
+	var tx evmclient.CommonTransaction
+	if activated, err := client.IsEIP1559Activated(); err != nil {
+		if activated {
+			tx = evmtransaction.NewDynamicFeeTransaction(cId, opts.Nonce.Uint64(), &p.BridgeAddress, big.NewInt(0), opts.GasTipCap, opts.GasFeeCap, opts.GasLimit, input)
+		} else {
+			tx = evmtransaction.NewTransaction(opts.Nonce.Uint64(), &p.BridgeAddress, big.NewInt(0), opts.GasLimit, opts.GasPrice, input)
+		}
```
Comment (waymobetta) [inline chains/evm/voter/proposal.go:160]: see previous comments above; this logic should live outside of the error block

File: chains/evm/evmclient/evm-client.go (hunk 1 above)
Comment (vezenovm) [reply chains/evm/evmclient/evm-client.go:133]: No it isn't I will switch it

File: chains/evm/voter/proposal.go (hunk 2 above)
Comment (vezenovm) [reply chains/evm/voter/proposal.go:112]: Same for these

**Comment by P1sar** [comment]
ill mere it in to your branch, since its hard to follow  changes from 2 branches
